include/fastdds/dds/topic/DataWriter.hpp

```diff
@@ -0,0 +1,188 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DataWriter.hpp
+ */
+
+#ifndef _FASTRTPS_DATAWRITER_HPP_
+#define _FASTRTPS_DATAWRITER_HPP_
+
+#include <fastdds/rtps/common/Time_t.h>
+#include <fastrtps/qos/DeadlineMissedStatus.h>
+#include <fastrtps/qos/IncompatibleQosStatus.hpp>
+
+namespace eprosima {
+namespace fastrtps{
+
+class WriterQos;
+class LivelinessLostStatus;
+class TopicAttributes;
+
+namespace rtps
+{
+class RTPSWriter;
+class RTPSParticipant;
+class WriteParams;
+class WriterAttributes;
+class InstanceHandle_t;
+class GUID_t;
+}
+
+} // namespace fastrtps
+
+namespace fastdds {
+namespace dds {
+
+class PublisherListener;
+class PublisherImpl;
+class Publisher;
+
+class TypeSupport;
+
+class DataWriterImpl;
+class DataWriterListener;
+
+/**
+ * Class DataWriter, contains the actual implementation of the behaviour of the DataWriter.
+ * @ingroup FASTDDS_MODULE
+ */
+class DataWriter
+{
+    friend class PublisherImpl;
+    friend class DataWriterImpl;
+
+    /**
+     * Create a data writer, assigning its pointer to the associated writer.
+     * Don't use directly, create Publisher using DomainRTPSParticipant static function.
+     */
+    DataWriter(
+            DataWriterImpl* impl);
+
+public:
+
+    virtual ~DataWriter();
+
+    /**
+     * Write data to the topic.
+     * @param data Pointer to the data
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data);
+
+    /**
+     * Write data with params to the topic.
+     * @param data Pointer to the data
+     * @param params Extra write parameters.
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data,
+            fastrtps::rtps::WriteParams& params);
+
+    /**
+     * Write data with handle.
+     * @param data Pointer to the data
+     * @param handle InstanceHandle_t.
+     * @return True if correct
+     * @par Calling example:
+     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     */
+    bool write(
+            void* data,
+            const fastrtps::rtps::InstanceHandle_t& handle);
+
+    /**
+     *
+     * @return
+     */
+    const fastrtps::rtps::GUID_t& guid();
+
+    fastrtps::rtps::InstanceHandle_t get_instance_handle() const;
+
+    /**
+     * Get topic data type
+     * @return Topic data type
+     */
+    TypeSupport get_type() const;
```

Comment:
According to DDS Standard, this function should be here. It should be in TypeSupport class

Reply:
Our `TypeSupport` isn't standard yet.